src/rust/engine/options/src/parse.rs

```diff
@@ -11,16 +11,20 @@ peg::parser! {
         rule whitespace() -> ()
             = quiet!{ " " / "\n" / "\r" / "\t" }
 
-        rule string() -> String
+        rule value<T>(parse_value: rule<T>) -> T
```

Comment:
See https://docs.rs/peg/latest/peg/#rule-parameters for how PEG parser rules can be parameterized by type, and take expression fragments as arguments.